### ISSUE TITLE
if replace DCN with normal conv, How much performance drop does it make?

### ISSUE THREAD

**Issue by zyc4me** [xingyizhou/CenterNet#64]
if replace DCN with normal conv, How much performance drop does it make?  such as resnet18,  AP 28.1 will drop to ??

**Comment by xingyizhou** [comment]
I haven't tried res 18. [DLAv0](https://github.com/xingyizhou/CenterNet/blob/master/src/lib/models/networks/dlav0.py) (The original DLA, without dcn and final upsample) is about 33 flip AP. You are welcomed to try the [vanilla resnet](https://github.com/xingyizhou/CenterNet/blob/master/src/lib/models/networks/msra_resnet.py) yourself and report the accuracy.

**Comment by Wondersui** [comment]
if you change DCNv2 to DCNv1, the AP will drop tp?

**Comment by xingyizhou** [comment]
@Wondersui I do have tried using DCN implementation from mmdet. Here DCNv1 and DCNv2 give the same performance. However, both implementations are slightly (< 1AP) worse than [CharlesShang's DCN implementation](https://github.com/CharlesShang/DCNv2). That's why we keep using this implementation. I am not sure if the difference is from randomness or details in the DCN implementation. To answer your question: the AP will drop at most 1.

**Comment by cwjhx** [comment]
> I haven't tried res 18. [DLAv0](https://github.com/xingyizhou/CenterNet/blob/master/src/lib/models/networks/dlav0.py) (The original DLA, without dcn and final upsample) is about 33 flip AP. You are welcomed to try the [vanilla resnet](https://github.com/xingyizhou/CenterNet/blob/master/src/lib/models/networks/msra_resnet.py) yourself and report the accuracy.

I will try DLAv0 （The original DLA, without dcn and final upsample），How train this model？Thank you.

**Comment by xingyizhou** [comment]
`--arch dlav0_34`

**Comment by cwjhx** [comment]
> `--arch dlav0_34`

Thank you.
but this code is error：networks/dlav0.py
def get_pose_net(num_layers, heads, add_conv=256, down_ratio=4):
  model = DLASeg('dla{}'.format(num_layers), heads,
                 pretrained=True,
                 down_ratio=down_ratio,
                 head_conv=head_conv)
  return model
it should be changed to :
def get_pose_net(num_layers, heads, head_conv=256, down_ratio=4):
  model = DLASeg('dla{}'.format(num_layers), heads,
                 pretrained=True,
                 down_ratio=down_ratio,
                 head_conv=head_conv)
  return model
？

**Comment by wangshankun** [comment]
> > I haven't tried res 18. [DLAv0](https://github.com/xingyizhou/CenterNet/blob/master/src/lib/models/networks/dlav0.py) (The original DLA, without dcn and final upsample) is about 33 flip AP. You are welcomed to try the [vanilla resnet](https://github.com/xingyizhou/CenterNet/blob/master/src/lib/models/networks/msra_resnet.py) yourself and report the accuracy.
> 
> I will try DLAv0 （The original DLA, without dcn and final upsample），How train this model？Thank you.

have any result？

**Comment by andriiD** [comment]
resnet18 with DCN 0.285. [vanilla resnet](https://github.com/xingyizhou/CenterNet/blob/master/src/lib/models/networks/msra_resnet.py) with the same training settings and ordinary conv layers 0.231

**Comment by dheerajpreddy** [comment]
Can you please upload the models trained without DCN? Thanks

**Comment by andriiD** [comment]
> Can you please upload the models trained without DCN? Thanks

Unfortunately I cannot share the model weights
